pypika/terms.py

```diff
@@ -701,10 +712,16 @@ def get_sql(self, **kwargs):
                 largest=self.largest,
                 smallest=self.smallest,
             ) if self.largest != self.smallest else self.largest
-        return 'INTERVAL \'{expr} {unit}\''.format(
-            expr=expr,
-            unit=unit,
-        )
+
+        if dialect == Dialects.MYSQL:
```

Comment:
instead of using an if branch for this whole part, you could use a dict to map the interval to a string template and then format it.